shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxBody.hs

```diff
@@ -415,45 +419,59 @@ deriving instance Eq (TxIn era)
 
 deriving instance Show (TxIn era)
 
-deriving instance (Era era) => NFData (TxIn era)
+deriving instance Era era => NFData (TxIn era)
 
 instance NoUnexpectedThunks (TxIn era)
 
 -- | The output of a UTxO.
 data TxOut era
   = TxOutCompact
       {-# UNPACK #-} !BSS.ShortByteString
-      {-# UNPACK #-} !Word64
+      !(Core.CompactForm (Core.Value era))
```

Comment:
One way to get around this would be to instead make `TxOut` compactible.

This would also potentially resolve the "non-representable coin" issue, since TxOuts should never have non-valid coins.